test/functional/JLM_Tests/src/org/openj9/test/java/lang/management/TestOpenJ9DiagnosticsMXBean.java

```diff
@@ -520,80 +473,75 @@ private static boolean findAndDeleteFile(String dirPath, String fileName) {
 		for (File f : files) {
 			found = f.getName().startsWith(fileName);
 			if (found) {
-				try {
-					reader = new BufferedReader(new FileReader(f));
-					reader.close();
-					boolean res = f.delete();	
-					Assert.assertTrue(res, "Failed to delete the file");
-					break;
-				} catch (java.io.FileNotFoundException e) {
-					Assert.fail("Unexpected FileNotFoundException : " + e.getMessage());
-				} catch (java.io.IOException e) {
-					Assert.fail("Unexpected IOException : " + e.getMessage());
-				}
+				reader = new BufferedReader(new FileReader(f));
+				reader.close();
+				boolean res = f.delete();
+				Assert.assertTrue(res, "Failed to delete the file");
+				break;
 			}
 		}
 		return found;
 	}
 
+	/**
+	 * Function to check if it is zOS and system dump agent.
+	 *
+	 * @param agent Indicates the dump agent.
+	 * @return a boolean indicating if it is zOS and system dump agent
+	 */
+	private static boolean isZOSSystemAgent(String agent) {
+		if ("z/OS".equalsIgnoreCase(os) && "system".equals(agent)) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+
 	/**
 	 * Internal function: Starts a remote server to connect to.
 	 */
-	private static void startRemoteServer() {
-		logger.info("Start Remote Server!");
+	private static void startRemoteServer() throws IOException {
+		logger.info("Starting Remote Server to Monitor!");
+		char fs = File.separatorChar;
+		/* Set up a randomly named file and pass this to child process. Parent keeps checking
+		* for its existence on the file-system. When this goes missing, it implies the child
+		* process is up and running.
+		*/
+		Random randomGenerator = new Random();
+		String tmpName = System.getProperty("java.io.tmpdir") + fs + randomGenerator.nextInt(10000) + ".lock";
```

Comment:
A better approach would be to use `File.createTempFile()` instead of hoping the random number won't relate to an existing lock file.

Reply:
Thanks Keith, made the changes.